Gems/MachineLearning/Code/Source/Algorithms/Activations.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Algorithms/Activations.h>
+#include <AzCore/Math/SimdMath.h>
+#include <AzCore/Math/MatrixMxN.h>
+
+namespace MachineLearning
+{
+    AZStd::vector<AZ::Edit::EnumConstant<ActivationFunctions>> GetActivationEnumValues()
+    {
+        AZStd::vector<AZ::Edit::EnumConstant<ActivationFunctions>> values;
+        values.emplace_back(ActivationFunctions::ReLU, "ReLU");
+        values.emplace_back(ActivationFunctions::Sigmoid, "Sigmoid");
+        values.emplace_back(ActivationFunctions::Softmax, "Softmax");
+        values.emplace_back(ActivationFunctions::Linear, "Linear");
+        return values;
+    }
+
+    void OneHotEncode(AZStd::size_t value, AZStd::size_t maxValue, AZ::VectorN& output)
+    {
+        AZ_Assert(value <= maxValue, "Requested one-hot encode of an out of range value");
+        output.Resize(maxValue);
+        output.SetZero();
+        output.SetElement(value, 1.0f);
+    }
+
+    AZStd::size_t ArgMaxDecode(const AZ::VectorN& vector)
+    {
+        const AZStd::size_t numElements = vector.GetDimensionality();
+        float maxValue = 0.0f;
+        AZStd::size_t maxIndex = 0;
+        for (AZStd::size_t iter = 0; iter < numElements; ++iter)
+        {
+            if (vector.GetElement(iter) > maxValue)
+            {
+                maxValue = vector.GetElement(iter);
+                maxIndex = iter;
+            }
```

Comment:
I don't know how expensive the call to `GetElement` is, but you can store the value the first time it is called.
```suggestion
            if (float currValue = vector.GetElement(iter); currValue > maxValue)
            {
                maxValue = currValue;
                maxIndex = iter;
            }
```